Tests/Realm.Tests/Database/InstanceTests.cs

```diff
@@ -463,6 +419,32 @@ public void Compact_WhenResultsAreOpen_ShouldReturnFalse()
             token.Dispose();
         }
 
+        [Test]
+        public void Compact_WhenShouldDeleteIfMigrationNeeded_PreservesObjects()
```

Comment:
This is the test for https://github.com/realm/realm-dotnet/issues/3583 and https://github.com/realm/realm-dotnet/issues/3612